### PR TITLE
183012754 optional login

### PR DESCRIPTION
**Pull request by alexosugo**
- Refactored login code to use new endpoints for staging. 
- Uses Retrofit as the main network client. 
- Changes login code to adhere to new app architecture. 
- Adds Login with Google on the Welcome screen. 
- Minor refactors of Welcome Fragment.

### CODE REVIEW

File: app/src/main/res/values-om/strings.xml [hunk 1]
```diff
@@ -5,10 +5,10 @@
     <string name="app_name" translatable="false">Stax</string>
     <string name="payment_root_url" translatable="false">https://stax.me/pay?%1$s</string>
     <string name="payment_url_end" translatable="false">%1$s-%2$d-%3$s-%4$d</string>
-    <string name="root_url" translatable="false">https://www.usehover.com</string>
```
Comment (davkutalek) [inline app/src/main/res/values-om/strings.xml:8]: These strings should not be duplicated in localization files, that is the point of the translatable flag

**Review by davkutalek — APPROVED** [review]
LGTM, just minor issue with the strings

**Review by androidmaven — APPROVED** [review]
LGTM.

Thanks!

File: app/src/main/res/values-om/strings.xml (hunk 1 above)
Comment (alexosugo) [reply app/src/main/res/values-om/strings.xml:8]: @davkutalek Yeah, I agree. I ran into them while working on this feature and removed some of them. I'll clear the rest before merging.

**Pull request merged**